### ISSUE TITLE
feat[train]: Add epoch metrics collector

### ISSUE THREAD

**Issue by Hamza-Bin-Aamir** [Hamza-Bin-Aamir/PIVOT#79]
Store per-epoch metrics (loss, accuracy, learning rate, etc.)
Store train and validation metrics separately
Timestamp each metric collection
Implement circular buffer for memory efficiency (keep last N epochs)
Support custom metric registration
